src/script/lua_api/l_inventory.cpp

```diff
@@ -273,18 +270,17 @@ int InvRef::l_set_lists(lua_State *L)
 	}
 
 	// Make a temporary inventory in case reading fails
-	Inventory *tempInv(inv);
-	tempInv->clear();
+	Inventory tempInv(inv->getItemDef());
 
 	Server *server = getServer(L);
 
 	lua_pushnil(L);
 	while (lua_next(L, 2)) {
 		const char *listname = lua_tostring(L, -2);
-		read_inventory_list(L, -1, tempInv, listname, server);
+		read_inventory_list(L, -1, &tempInv, listname, server);
 		lua_pop(L, 1);
 	}
-	inv = tempInv;
+	*inv = tempInv;
```

Comment:
Yes, that's the whole point.  Previously it just set the address in the `inv` pointer, which promptly went out of scope.
The only reason it worked at all is because the Inventory pointer was just copied as `inv`, instead of actually creating a separate Inventory object.